File: 6KYU/ASCII Fun #2: Funny Dots/solution.cs
using System.Linq;
public class Kata
{
    public static string Dot(int w, int h)
    {
        var row = $"\n|{string.Join("|", Enumerable.Repeat(" o ", w))}|\n";
        var splitter = $"+{string.Join("+", Enumerable.Repeat("---", w))}+";
        return $"{splitter}{string.Join($"{splitter}", Enumerable.Repeat(row, h))}{splitter}";
    }
}

File: 6KYU/ASCII Fun #2: Funny Dots/solution.go
package kata

import "strings"

func row(w int) string {
	s := []string{}
	for i := 0; i < w; i++ {
		s = append(s, " o ")
	}
	return "|" + strings.Join(s, "|") + "|"
}

func seperator(w int) string {
	s := []string{}
	for i := 0; i < w; i++ {
		s = append(s, "---")
	}
	return "+" + strings.Join(s, "+") + "+"
}

func Dot(w, h int) string {
	rows := []string{}
	rows = append(rows, seperator(w))
	for i := 0; i < h; i++ {
		rows = append(rows, row(w))
		if h-i > 1 {
			rows = append(rows, seperator(w))
		}
	}
	rows = append(rows, seperator(w))
	return strings.Join(rows, "\n")
}
